client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/services/LibFreeRDP.java

```diff
@@ -273,6 +275,61 @@ public static boolean setConnectionInfo(int inst, BookmarkBase bookmark) {
         String[] arrayArgs = args.toArray(new String[args.size()]);
         return freerdp_parse_arguments(inst, arrayArgs);
     }
+    
+    public static boolean setConnectionInfo(int inst, Uri openUri) {
+        ArrayList<String> args = new ArrayList<String>();
+        HashMap<String, String> map = new HashMap<String, String>();
```

Comment:
I realize it might be not necessary. The parse argument function actually already support overwrite for same argument key.
I will do more test to double check